### PR TITLE
Feature/LUM 2 - demo docker container

### PR DESCRIPTION
**Pull request by tlhampton13**
Implemented a demo docker container that is fully self-contained.

### CODE REVIEW

File: docker/demo/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Lumify Demo Docker Container
+The Lumify Demo Docker container encapsulates a minimal deployment of Lumify.  The container is self contained and can be run independant of the development environment, after it is built.
```
Comment (mrburrito) [inline docker/demo/README.md]: independ`e`nt

File: docker/demo/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	- [Docker](https://www.docker.com/)
+- HTML5 compliant web browser
+
+## Installation
+Execute the following commands to build the Lumify Demo Docker Container.
+
+```sh
+$ boot2docker init -m 8192
+$ boot2docker start 
+$ cd ~
```
Comment (mrburrito) [inline docker/demo/README.md:80]: Need to indicate that boot2docker is only required for OS X, not Linux and that `boot2docker init` only needs to be run once. We should also include the shell variable export command, or is that not necessary for the demo VM?

File: docker/demo/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+```sh
+$ ssh -p 2022 -i <working dir>/lumify/docker/demo/keys/id_rsa root@lumify-demo
+```
+
+## Redeploy Lumify to a running Docker container
+Execute the following commands to rebuild and deploy the Lumify web application to the currently running Docker container.
+
+```sh
+$ ~/lumify/docker/deploy-lumify-demo.sh
```
Comment (mrburrito) [inline docker/demo/README.md]: Let's rename script to `update-lumify-demo.sh`

File: docker/demo/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+```
+http://lumify-demo:8080
+```
+
+## ssh into lumify-demo
+After running the Lumify Docker Container you can 'ssh' into the instance using the following command.
+
+```sh
+$ ssh -p 2022 -i <working dir>/lumify/docker/demo/keys/id_rsa root@lumify-demo
+```
```
Comment (mrburrito) [inline docker/demo/README.md:119]: Can we make this a simple script that identifies the working directory and knows the relative path to the key so users don't have to type everything? Should this be a separate ticket?

File: docker/demo/README.md (hunk 2 above)
Comment (tlhampton13) [inline docker/demo/README.md:80]: The sell variable export is done in the build-demo.sh and run-demo.sh script.  You would not need to export them manually unless you want to run docker commands.  I'll add a section on how to set them anyway, someone might need it.

File: web/war/pom.xml [hunk 5]
```diff
@@ -206,5 +206,230 @@
             </plugins>
             </build>
         </profile>
+        <profile>
+            <id>web-war-with-ui-plugins</id>
```
Comment (mrburrito) [inline web/war/pom.xml:210]: Change profile ID to `with-ui-plugins`; also -- is this all the UI plugins, a basic set, specifically for the demo container?

Comment (mrburrito) [inline web/war/pom.xml:210]: Actually, strike that last change. Makes it obvious its for the war when building from the top level. Leave id as-is.

File: docker/demo/README.md (hunk 4 above)
Comment (tlhampton13) [inline docker/demo/README.md:119]: Separate ticket.

File: docker/demo/README.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+*\* The Boot2Docker initialization step only needs to be run once.*
+
+####2. Start the Boot2Docker Virtual Machine
+This command will start the Boot2Docker virtual machine within VirtualBox.
+
+```sh
+$ boot2docker start
+```
+
+*\* The Boot2Docker virtual machine only needs to be started if it has not been started or has been stoped.*
```
Comment (mrburrito) [inline docker/demo/README.md]: stopped

File: docker/demo/README.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+####2. Build the Lumify Docker Container
+Executing the build script will complile the Lumify software and build the Docker container hosting a minimal deployment of Lumify.  This step may take a significant amount of time to complete.
+
+```sh
+$ cd ~/lumify/docker
+$ ./build-demo.sh
+```
+
+## Running the demo
+Execute the following commnd to run the Lumify Demo Docker Container.
```
Comment (mrburrito) [inline docker/demo/README.md]: command

File: docker/demo/README.md [hunk 8]
```diff
@@ -0,0 +1,138 @@
+# Lumify Demo Docker Container
```
Comment (mrburrito) [inline docker/demo/README.md:1]: Node.js, npm and grunt-cli are also required dependencies. We need to add documentation indicating either where to find them or how to install them.

**Comment by mrburrito** [comment]
This is somewhat OBE and will be replaced by a Vagrant config.

**Pull request merged**